### PR TITLE
feat: add new blogpost "Getting Started: Gatsby vs. Next.js vs. Remix"

### PR DESCRIPTION
**Pull request by hofmann-felix**
Preview URL: https://satellytescommain21751-featblogframeworkcomparison.gtsb.io/blog/getting-started-gatsby-next-remix/

This PR adds a blog post comparing the frameworks Gatsby, Next.js and Remix. The focus is set on the development process and the different concepts behind these frameworks. 

Another article is planned in the future, in which the performance etc of these frameworks will be compared.

### CODE REVIEW

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+> ⚡ The options shown below for loading data only works in page/template components (they are bound to a certain url path or directory). To load data in components that do not have their own route, you must use other methods
+
+
+#### **Gatsby**
+
+A static generated page in Gatsby consists of two parts: A GraphQL query and the actual React component (here `BlogPage`), which must be exported as default. In the GraphQL query, the required data is fetched. In this example, the Gatsby plugin [gatsby-transformer-remark](https://www.gatsbyjs.com/plugins/gatsby-transformer-remark/) is used to get the content of the corresponding markdown file.
+In the actual component, the results of the query can then be accessed via `data`.
+
+In our Gatsby blog project, we created a file named `{markdownRemark.frontmatter__path}.tsx` to create a route for each blogpost. The important thing here is that the filename is the path of the query in curly brackets. This way the data from the markdown files determines the routes of the project.
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: I don't understand this. Where can I see that file creation? You only list the template below ?

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+With static site generation (SSG), the HTML is generated once at built time and reused for each request. Of the three frameworks in this article, only Remix does not support SSG.
+
+
+> ⚡ The options shown below for loading data only works in page/template components (they are bound to a certain url path or directory). To load data in components that do not have their own route, you must use other methods
+
+
+#### **Gatsby**
+
+A static generated page in Gatsby consists of two parts: A GraphQL query and the actual React component (here `BlogPage`), which must be exported as default. In the GraphQL query, the required data is fetched. In this example, the Gatsby plugin [gatsby-transformer-remark](https://www.gatsbyjs.com/plugins/gatsby-transformer-remark/) is used to get the content of the corresponding markdown file.
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: You talk about `BlogPage` but I see `BlogPost` in the markup ?

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+            html
+        }
+    }
+`
+
+export default BlogPost
+```
+
+
+> 💡 Alternatively, pages can also be created in `gatsby-node.js`. For this, the `createPages` API provided by Gatsby must be used. For more information on this, as well as an easy-to-understand example, see the [Gatsby documentation](https://www.gatsbyjs.com/docs/creating-and-modifying-pages/#creating-pages-in-gatsby-nodejs).
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: This is explicit about manually create pages with `createPages` vs implicit (gatsby page plugin). I think you should write a paragraph about this differentiation because it's pretty confusing here. Maybe it's related to my earlier confusion about the template you display itself ? (see other comment)

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+
+
+#### **Next.js**
+
+To create a static generated page in Next.js the `async` function `getStaticPaths` has to be exported. `getStaticPaths` must return an object with the keys:
+
+- `path`: This key determines which path will be pre-rendered. In the in-code comment below the structure of `path` is given as an example.
+- `fallback`: You can set the value of fallback to `true`, `false` or `blocking`
+    - `false` → Any paths not returned by `getStaticPaths` will end up in a 404 page
+    - `true` → On the first request to a not generated path, Next.js will serve a fallback version of the page, instead of a 404 page. The HTML and JSON of this path is now generated in the background and and then the JSON is sent to the requesting browser. Now the fallback page changes into the full page. Next.js will add this path to the pre-rendered pages in order to be able to provide the already generated page for further requests.
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: First I thought, wow so many details but after reading, this is a super important difference to gatsby isn't it? I will look for that fact if you provide some summary or conclusion 🤞 It might be worth to change the format as it looks very technical while it deserves a proper story.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+      This is useful if you have a page with a lot of static pages because then not all pages have to be generated at built time, which makes the built much faster.
+    - `blocking` → Same procedure as for `fallback: true`, except that there is no fallback while the HTML is being generated. There are only very special uses cases in which this makes sense, such as AMP.
+
+Remember to name the file after the relevant key from the `params` object with square brackets, in our case `[path].tsx`.
+To load the data for a blogpost the function `getStaticProps` is needed. This function now receives `{params}`, which can be used to determine the path and the depending data. Finally, in the actual page component, the data can be accessed via props (here via `{postData}`).
+
+```tsx
+export async function getStaticPaths() {
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md:157]: Does this happen in a specific file to name?

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+- `fallback`: You can set the value of fallback to `true`, `false` or `blocking`
+    - `false` → Any paths not returned by `getStaticPaths` will end up in a 404 page
+    - `true` → On the first request to a not generated path, Next.js will serve a fallback version of the page, instead of a 404 page. The HTML and JSON of this path is now generated in the background and and then the JSON is sent to the requesting browser. Now the fallback page changes into the full page. Next.js will add this path to the pre-rendered pages in order to be able to provide the already generated page for further requests.
+      This is useful if you have a page with a lot of static pages because then not all pages have to be generated at built time, which makes the built much faster.
+    - `blocking` → Same procedure as for `fallback: true`, except that there is no fallback while the HTML is being generated. There are only very special uses cases in which this makes sense, such as AMP.
+
+Remember to name the file after the relevant key from the `params` object with square brackets, in our case `[path].tsx`.
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: Should I look at the source code? Then write this afterwards. In addition I can't find the mentioned snippet, where should the reader look at ?

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+
+```tsx
+export async function getStaticPaths() {
+    //     paths: [
+    //         {params: {path: 'first-blogpost'}},
+    //         {params: {path: 'second-blogpost'}},
+    //         ...
+    //         {params: {path: 'fifth-blogpost'}}
+    //     ],
+    const paths = getAllPostPaths()
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md:165]: I'm confused. Should I look at the comment or ignore it? What is getAllPostPaths about ?

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+}
+```
+
+
+> 💡 The object returned by `getStaticProps` can include other keys besides `props`, such as `revalidate`. `revalidate` is set to `false` by default, but can also contain the amount in seconds after which a page re-generation can occur. In this way, individual static pages can be changed at a certain interval after being built. This is called [Incremental Static Regeneration (ISR)](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration).
+
+
+#### **Remix**
+
+Remix supports only server side rendering. An example of this can be found in the section below.
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: That's an important distinction. In addition I wonder if Next.js did that in the beginning too (only SSR) and later supported SSG) ? This would help to sharpen the image.

Reading this I think some kind of table would help to compare without knowing the exact content. Maybe you have an idea.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+
+The command `npx create-next-app@latest` can additionally be extended with the flag `--ts` to add direct TypeScript support to a new Next.js project. The following setup process does not provide any customization options apart from the project name.
+
+#### Remix
+
+A Remix project can be set up with the command `npx create-remix@latest`. Then you can specify the project name, choose between different host options (e.g. Netlify or Vercel) and between TypeScript and JavaScript.
+
+## Static Site Generation
+
+With static site generation (SSG), the HTML is generated once at built time and reused for each request. Of the three frameworks in this article, only Remix does not support SSG.
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md:50]: For Gatsby there is no running process involved (if you are not opt-in to SSR) and Next.js can't be run without the process running, so you can deploy to S3 and such. Is this correct? That would help here to explain as it's not clear what keeps running after the build.

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 2 above)
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: I think you should mention "since v4" as this is a brand new feature

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+
+Remix supports only server side rendering. An example of this can be found in the section below.
+
+## Server Side Rendering (SSR)
+
+With server side rendering (SSR), the HTML is rebuilt for each request. All three frameworks support SSR.
+
+#### **Gatsby**
+
+Besides static site generation (SSG) you can also use server side rendering (SSR) for certain use cases in Gatsby. For this, the function `getServerData` must be built into a page, in which the data is requested from the server. This data can then be accessed in the actual page component with `serverData`. In the following the example from the [Gatsby documentation](https://www.gatsbyjs.com/docs/how-to/rendering-options/using-server-side-rendering/) is considered:
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: Thanks for telling, I always wondered how that's working and I did not consult the docs before 💫 Being a naive reader I wonder about the server process. before I run gatsby only in CI/CD. For SSR I need to keep it running. So will I start & build once in CI/CD and then mount the gatsby server in my cloud environment a second time but keep it running (and it will NOT build everything?)

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 11]
```diff
@@ -0,0 +1,306 @@
...
+const SSRPage = ({ serverData }) => (
+  <main>
+    <h1>SSR Page with Dogs</h1>
+    <img alt="Happy dog" src={serverData.message} />
+  </main>
+)
+
+export default SSRPage
+
+export async function getServerData() {
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md:228]: How would I merge static and dynamic data? Or is this mutual exclusive? Say one dataset I can read via graphql and then it might update later. Is this covered? Does graphql play any role with SSR anymore ?

Comment (georgiee) [reply blog-posts/getting-started-gatsby-next-remix.md:228]: Those are questions I ask because of my interest. You can decide if there is some answer that you might want to include but it's not something I feel missing as you get the point of SSR.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 12]
```diff
@@ -0,0 +1,306 @@
...
+
+To use SSR in Next.js you have to use `getServerSideProps` instead of `getStaticProps`. This function must load the data and return it, too. The data can then be accessed in the page component (in the example via `data`). The following example shows a simple SSR implementation from the [Next.js documentation](https://nextjs.org/docs/basic-features/pages#server-side-rendering).
+
+```tsx
+function Page({ data }) {
+  // Render data...
+}
+
+// This gets called on every request
+export async function getServerSideProps() {
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md:261]: I like the naming when you compare it with the SSR version `getStaticProps` 👌 That's something you could remind the reader about in the text too.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 13]
```diff
@@ -0,0 +1,306 @@
...
+
+export default function PostSlug() {
+    const post = useLoaderData();
+    return (
+            <div dangerouslySetInnerHTML={{ __html: post.html }} />
+    );
+}
+```
+
+## Adding Page Metadata
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md:295]: Interesting aspect that you highlight. Thanks.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 14]
```diff
@@ -0,0 +1,306 @@
...
+Next.js provides a `head` component that can be built into any page to add metadata.
+
+```tsx
+//...
+import Head from "next/head";
+
+const Home: NextPage = () => {
+    return (
+        <Layout>
+            <Head>
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md:346]: I wonder if they just encapsulate what helmet already does ?

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 15]
```diff
@@ -0,0 +1,306 @@
...
+    const post = useLoaderData();
+    return (
+            <div dangerouslySetInnerHTML={{ __html: post.html }} />
+    );
+}
+```
+
+## Adding Page Metadata
+
+The page meta data, such as the title or description, are especially important for SEO. In the following examples, a title should be added to the pages. For this, each framework offers its own solution.
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]:
```suggestion
The page header data, such as the html title, meta description and structural data like [open graph](https://ogp.me/) are especially important for SEO. In the following examples, a title should be added to the pages. For this, each framework offers its own solution.
```

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 16]
```diff
@@ -0,0 +1,306 @@
...
+
+```tsx
+//...
+import {MetaFunction} from "remix";
+
+export const meta: MetaFunction = () => {
+    return {
+        title: "Remix Blog",
+    };
+};
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: The title is easy I wonder how a meta tag (description or open graph) would be added right now.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 17]
```diff
@@ -0,0 +1,306 @@
...
+    <Layout>
+	      {/*content goes here*/}
+    </Layout>
+  );
+}
+```
+
+## Conclusion
+
+Next.js and Gatsby have quite a lot in common as they support both SSR and SSG. The differences are more in detail such as incremental static generation in Next.js or incremental builds in Gatsby. However, it is noticeable that SSR in Gatsby is still a relatively new and not as developed feature as it is in Next.js. The roots of Next.js are in SSR, but relatively soon they also supported SSG. Thus, this framework supports both approaches well.
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: A conclusion should never contain more information. Here I finally read about the fact SSG is new to Gatsby and that Next.js introduced it after being a SSR framework. Those facts deserve their place in the article. The conclusion itself reads very well though!

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 18]
```diff
@@ -0,0 +1,306 @@
...
+
+Remix is the newest of these three frameworks and only supports SSR. However, with the appropriate headers, SSG is approached very closely, which means that it may be possible to completely replace real SSG. The developers of Remix have published an interesting [video](https://www.youtube.com/watch?v=bfLFHp7Sbkg) about this.
+
+![blog-overview.png](images/gastby-next-remix-screenshot-twitter.png)
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: I think that's a neat message but focus on the other side of the story:

1. You either manage a running server process plus caching layer and save spend time on optimizing them (SSR)
2. Or you save the time to manage the server and use probably the equal amount of time into your build pipeline where you will struggle to build thousand and thousand of pages at once (SSG).

There is simply no shortcut. I think SSG had its glamour, but I guess we are going full circle and come back to SSR soon. Gatsby introducing SSR capabilities is a first sign of that.

Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md]: This is a rather abrupt ending. Some personal words like "Thanks for reading and I hope you got some context for your decision to pick one of the technologies" will help here (phrase your own please to get your personal touch in there)

**Review by georgiee — CHANGES_REQUESTED** [review]
This reads very well with a super clear storyline introducing the topic, looking at SSG, then SSR and metadata in the end. The conclusion reads great too.

I have provided a few comments while I read it. I did not check for typos and such to focus on the story. I think there is really not much to do, just some polishing opportunities.

I also love the teaser image you picked ✨ There are actually no blocking issues but things that would help the reader so maybe you can give me some feedback and we can then proceed with any state that you like. I would also appreciate if someone else proof read it or did this already happen before this PR ?


Thank you a lot for this great blog post. Very much appreciated and it hits the right time after Remix got open source recently 💯

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 1 above)
Comment (hofmann-felix) [reply blog-posts/getting-started-gatsby-next-remix.md]: I have reworded the sentence with the explanation and also included an example with the create-pages API. I hope that it is now easier to understand

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 3 above)
Comment (hofmann-felix) [reply blog-posts/getting-started-gatsby-next-remix.md]: I've added an example with the create-pages API too 👍

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 5 above)
Comment (hofmann-felix) [reply blog-posts/getting-started-gatsby-next-remix.md:157]: It is important that the filename and the name of the path-determining key match. I have now emphasized this in an additional sentence

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 6 above)
Comment (hofmann-felix) [reply blog-posts/getting-started-gatsby-next-remix.md]: Which mentioned snippet do you mean?

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 7 above)
Comment (hofmann-felix) [reply blog-posts/getting-started-gatsby-next-remix.md:165]: According the comment: 
In the explanation of the path array above, a reference to the comment is given. I'm not sure how to additionally show the reader that the comment is relevant.

According getAllPostPaths:
Thanks for pointing that out 🙏 There was definitely a missing explanation. I have added it.

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 16 above)
Comment (hofmann-felix) [reply blog-posts/getting-started-gatsby-next-remix.md]:
```
import type { MetaFunction } from "remix";

export const meta: MetaFunction = () => {
  return {
    title: "Josie's Shake Shack", // <title>Josie's Shake Shack</title>
    description: "Delicious shakes", // <meta name="description" content="Delicious shakes">
    "og:image": "https://josiesshakeshack.com/logo.jpg" // <meta property="og:image" content="https://josiesshakeshack.com/logo.jpg">
  };
};
```

This is the example from the Remix documentation. Would you like to see a more detailed example in the blogpost than just the html title? My thought behind this was to keep the examples as simple and understandable as possible.

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 17 above)
Comment (hofmann-felix) [reply blog-posts/getting-started-gatsby-next-remix.md]: I put the additional information form the conclusion in the paragraphs above 👍

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 14 above)
Comment (hofmann-felix) [reply blog-posts/getting-started-gatsby-next-remix.md:346]: I am not quite sure, the syntax looks the same in any case. I did a quick research and only found this: https://stackoverflow.com/questions/61063193/react-helmet-or-next-head-for-next-js-project#:~:text=As%20far%20as%20I%20know,if%20you%20are%20using%20Next.

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 4 above)
Comment (hofmann-felix) [reply blog-posts/getting-started-gatsby-next-remix.md]: Do you have a suggestion on how to make it not so "technical"?

**Comment by hofmann-felix** [comment]
@georgiee Thank you for your detailed review 🙏 It was very helpful, especially because as a creator of an article you lose the overview quickly and can't put yourself into the perspective of a reader (at least that's always the case for me 😅).  

I have added most of your suggestions/questions to the blog post or commented on them. A few aspects (the comments from you that I have not commented on or marked with "👍") still have to be addressed, as I have to do a bit of research for it. 

FYI: Fabian has already read the article before this PR.

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 17 above)
Comment (georgiee) [reply blog-posts/getting-started-gatsby-next-remix.md]: I have seen that, thank you 🙌

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 16 above)
Comment (georgiee) [reply blog-posts/getting-started-gatsby-next-remix.md]: oh that's surprising. They map description to meta description (name notation) and title to the html title while the og tag is mapped to the meta `property` notation. This is quite some magic happening here isn't it ? I think that example is much better. 

A simple title property can't tell me how on earth it would allow me to create meta tags. Especially compared to the straight forward and magic-less approaches from the other frameworks.

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 4 above)
Comment (georgiee) [reply blog-posts/getting-started-gatsby-next-remix.md]: No. I though about writing it as a text without the bullets, but I don't know if that helps or hurt. Keep it like that then. Thanks for thinking about it 🙏

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 9 above)
Comment (feedm3) [reply blog-posts/getting-started-gatsby-next-remix.md:50]: If you only have static pages in next you can run `next export`. Would be the same as `gatsby build`.

https://nextjs.org/docs/advanced-features/static-html-export#unsupported-features

My guess is that most projects won't do this, as Next gives you so much more features with the running process.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 19]
```diff
@@ -0,0 +1,361 @@
...
+
+export default function Post({postData}: { postData: PostData }) {
+    return (
+            <div dangerouslySetInnerHTML={{__html: postData.contentHtml}}/>
+    )
+}
+```
+
+
+> 💡 The object returned by `getStaticProps` can include other keys besides `props`, such as `revalidate`. `revalidate` is set to `false` by default, but can also contain the amount in seconds after which a page re-generation can occur. In this way, individual static pages can be changed at a certain interval after being built. This is called [Incremental Static Regeneration (ISR)](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration).
```
Comment (feedm3) [inline blog-posts/getting-started-gatsby-next-remix.md]: Maybe you can also add this sentence:

```
This technique is also known as `stale-while-revalidate`.
```

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 8 above)
Comment (feedm3) [reply blog-posts/getting-started-gatsby-next-remix.md]: Remix explicitly only does SSR, because they have an architecture in mind where SSR is superior or equal to SSG in all manners.

Some of their reasons can be found in their latest article about "Remix vs Next". 
https://remix.run/blog/remix-vs-next

tl;dr:
- CDN with proper caching headers will always be fast, except for cold requests (like the first request)
- If the backend is so slow that cold requests are a problem, the backend should be improved or cached
- If pages are visited so rarely that they are always cold, it's a business problem and not a technical

Comment (feedm3) [reply blog-posts/getting-started-gatsby-next-remix.md]: We should add this to this part. Otherwise it looks like it's a problem or big downside.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 20]
```diff
@@ -0,0 +1,361 @@
...
+
+
+#### **Remix**
+
+Remix supports only server side rendering. An example of this can be found in the section below.
+
+## Server Side Rendering (SSR)
+
+With server side rendering (SSR), the HTML is rebuilt for each request. All three frameworks support SSR.
```
Comment (feedm3) [inline blog-posts/getting-started-gatsby-next-remix.md]:
```
With server side rendering (SSR), the HTML is built on each request. All three frameworks support SSR. 

Usually, you use SSR in combination with a CDN to improve performance and responsiveness.
```

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 11 above)
Comment (feedm3) [reply blog-posts/getting-started-gatsby-next-remix.md:228]: Yes, you can mix it up: https://www.gatsbyjs.com/docs/reference/rendering-options/server-side-rendering/#interplay-with-build-time-graphql-queries

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 21]
```diff
@@ -0,0 +1,361 @@
...
+export async function getServerData() {
+  try {
+    const res = await fetch(`https://dog.ceo/api/breeds/image/random`)
+
+    if (!res.ok) {
+      throw new Error(`Response failed`)
+    }
+
+    return {
+      props: await res.json(),
```
Comment (feedm3) [inline blog-posts/getting-started-gatsby-next-remix.md:237]: The `cache-control` header is crucial when doing SSR, therefore I would add it to the example:

```
{
	props: await res.json(),
	headers: {
      // cache the page in browser for 10 seconds, on the CDN for 1 minute and keep it stale for additional 4 mintues
      'Cache-Control': 'public, max-age=10, s-maxage=60, stale-while-revalidate=240',
    }, 
}
```

Comment (feedm3) [reply blog-posts/getting-started-gatsby-next-remix.md:237]: Ok, there is also no `Cache-Control` header in the next example. We should leave it out then and handle this topic in a different blog post.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 22]
```diff
@@ -0,0 +1,361 @@
...
+    </Layout>
+  );
+}
+```
+
+## Conclusion
+
+Next.js and Gatsby have quite a lot in common as they support both SSR and SSG. The differences are more in detail such as incremental static generation in Next.js or incremental builds in Gatsby. However, it is noticeable that SSR in Gatsby is still a relatively new and not as developed feature as it is in Next.js. 
+
+Remix is the newest of these three frameworks and only supports SSR. However, with the appropriate headers, SSG is approached very closely, which means that it may be possible to completely replace real SSG. The developers of Remix have published an interesting [video](https://www.youtube.com/watch?v=bfLFHp7Sbkg) about this.
```
Comment (feedm3) [inline blog-posts/getting-started-gatsby-next-remix.md]: The developers of Remix have published an interesting [video](https://www.youtube.com/watch?v=bfLFHp7Sbkg) 
and [blog post](https://remix.run/blog/remix-vs-next) about this.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 23]
```diff
@@ -0,0 +1,361 @@
...
+
+![blog-overview.png](images/gastby-next-remix-screenshot-twitter.png)
+
+Thanks for reading my article about the different frameworks. I hope you learned something about the basic concepts behind Gatsby, Next.js and Remix. As I mentioned in the introduction, there is no clear answer to the question "Which framework should be used in my next project?" because every project has its individual requirements. Maybe now you know which framework can implement certain requirements well.
```
Comment (feedm3) [inline blog-posts/getting-started-gatsby-next-remix.md]: Not sure if the last sentence is good. We don't judge the frameworks, so there is not really an indicator of which framework can be used.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 24]
```diff
@@ -0,0 +1,361 @@
...
+}
+```
+
+## Conclusion
+
+Next.js and Gatsby have quite a lot in common as they support both SSR and SSG. The differences are more in detail such as incremental static generation in Next.js or incremental builds in Gatsby. However, it is noticeable that SSR in Gatsby is still a relatively new and not as developed feature as it is in Next.js. 
+
+Remix is the newest of these three frameworks and only supports SSR. However, with the appropriate headers, SSG is approached very closely, which means that it may be possible to completely replace real SSG. The developers of Remix have published an interesting [video](https://www.youtube.com/watch?v=bfLFHp7Sbkg) about this.
+
+![blog-overview.png](images/gastby-next-remix-screenshot-twitter.png)
```
Comment (feedm3) [inline blog-posts/getting-started-gatsby-next-remix.md]: We should link to the source here.

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 16 above)
Comment (hofmann-felix) [reply blog-posts/getting-started-gatsby-next-remix.md]: Yes definitely makes sense. I have now added `description` and `og:image` to the examples and explained this in a bit more detail in Remix

**Comment by feedm3** [comment]
One last thing: Remix has one SSR functionality that the other frameworks don't have. Resource routes. Remix can return any data from an URL, it doesn't have to be a page. 

First thought on how to write it:
```
Remix

... {your text}

Other than Next and Gatsby, Remix can't just return components from the SSR routes. Remix can also return other data, like images or PDF. This feature is called "Resouce routes". You can read more about this in the [official documentation](https://remix.run/docs/en/v1/guides/resource-routes). 
```

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 25]
```diff
@@ -0,0 +1,381 @@
...
+To add page metadata in Gatsby, the `gatsby-plugin-react-helmet` and `React Helmet` are recommended. Generally, Gatsby plugins are installed via a package manager and then have to be added to `gatsby-config.js`.
+So in our example blog we first had to run `npm install gatsby-plugin-react-helmet react-helmet` and then add the plugin to `gatsby-config.js`:
+
+```jsx
+{
+  plugins: [`gatsby-plugin-react-helmet`]
+}
+```
+
+After that, you can access the `React Helemet` within the JSX of a component, like for example in the index page.
```
Comment (feedm3) [inline blog-posts/getting-started-gatsby-next-remix.md]: Typo.

`React Helmet`

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 16 above)
Comment (feedm3) [reply blog-posts/getting-started-gatsby-next-remix.md]: There is very little magic involved: https://remix.run/docs/en/v1/api/conventions#meta

> There are a few special cases like `title` renders a `<title>` tag, `og:style` tags will render `<meta property content>`, the rest render `<meta name={key} content={value}/>`.

I just learned that there is basically no difference in `meta name` vs `meta property`: https://stackoverflow.com/questions/22350105/whats-the-difference-between-meta-name-and-meta-property

Comment (georgiee) [reply blog-posts/getting-started-gatsby-next-remix.md]: Thanks for the changes. Looks much better.

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 18 above)
Comment (georgiee) [reply blog-posts/getting-started-gatsby-next-remix.md]: As spoken, let's remove the twitter screenshot otherwise the conclusion is great!

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 1 above)
Comment (georgiee) [reply blog-posts/getting-started-gatsby-next-remix.md]: Thanks for the chat, that's clear now: It's a mechanic we never used so I was confused.

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 6 above)
Comment (georgiee) [reply blog-posts/getting-started-gatsby-next-remix.md]: You told me the details and we concluded that a highlight in the code through prismjs would help a lot when talking about code.

File: blog-posts/getting-started-gatsby-next-remix.md (hunk 8 above)
Comment (georgiee) [reply blog-posts/getting-started-gatsby-next-remix.md]: This was fixed. Thanks for the text.

File: blog-posts/getting-started-gatsby-next-remix.md [hunk 26]
```diff
@@ -184,6 +185,16 @@ export default function Post({postData}: { postData: PostData }) {
 }
 ```
 
+`getStaticPaths` must return an object with the keys:
```
Comment (georgiee) [inline blog-posts/getting-started-gatsby-next-remix.md:188]: much better here after the code snippet

**Review by georgiee — APPROVED** [review]
Let's merge this as soon as you are ready (update date, change single author) 👍 Thanks for your efforts here, very much appreciated ✨
